src/core/webview/ClineProvider.ts

```diff
@@ -1371,7 +1372,11 @@ export class ClineProvider extends EventEmitter<ClineProviderEvents> implements
 		} = await this.getState()
 
 		const machineId = vscode.env.machineId
-		const allowedCommands = vscode.workspace.getConfiguration(Package.name).get<string[]>("allowedCommands") || []
+		const allowedCommandsState = allowedCommands || []
+		const allowedCommandsWorkspace =
+			vscode.workspace.getConfiguration(Package.name).get<string[]>("allowedCommands") || []
+		const allowedCommandsCombined = [...new Set([...allowedCommandsState, ...allowedCommandsWorkspace])]
```

Comment:
Since this is shared with Roo, maybe wrap this in `//kilocode_change start/end` comment? But it looks like it might be a bug in Roo too, so maybe we could open this PR against them instead?